src/handlers/status-retriever-handler.ts

```diff
@@ -75,7 +75,7 @@ function validateEvent(userId: string) {
   return trimmedUserId;
 }
 
-function transformResponseFromDynamoDatabase(item: FullAccountInformation) {
+function transformResponseFromDynamoDatabase(item: Partial<FullAccountInformation>) {
```

Comment:
Question: is this why the in the interface, the history has been changed to a required field, by utilising this Partial?

Reply:
Thank you for reviewing @liamquayle 🙏  
yes - essentially I needed an interface that would describe what we are expecting the get back from the DynamoDB when we get the full row for a given user. Given our implementation, if the user has a record in the DB, then we expect some fields to be present for sure (key field, the status flags, some timestamps etc). So I set those field as non-optional in the interface. But I then wrapped parameter type that this function take into a Partial, so that I get back a "copy" of that interface where all fields have been marked as optional, so that we can still use it the way it was being used here :)